feat: public and private routes implemented

DaniilShuliyatev opened the pull request:
move mockImages to publick path
move getDisplayString function from PlayerCard component to shared/utils path implemented the functions of dynamically displaying the active tab in Navbar and StatisticsPage, depending on the current location.pathname implemented new routes: login, signup, statistics_team, statistics_players, create_team divided the content from the StatisticsPage component into the StatisticsPage and FantasyTeamPage components added blank page components: SinupPage, LoginPage, EditTeamPage, CreateTeamPage

DaniilShuliyatev committed:
feat: public and private routes implemented
move mockImages to publick path
move getDisplayString function from PlayerCard component to shared/utils path
implemented the functions of dynamically displaying the active tab in Navbar and StatisticsPage, depending on the current location.pathname
implemented new routes: login, signup, statistics_team, statistics_players, create_team
divided the content from the StatisticsPage component into the StatisticsPage and FantasyTeamPage components
added blank page components: SinupPage, LoginPage, EditTeamPage, CreateTeamPage

Yauhenia-Zhdanovich commented on frontend/src/pages/editTeamPage/EditTeamPage.style.ts:
```diff
@@ -0,0 +1,10 @@
+import { styled, Typography } from "@mui/material";
+
+export const CustomHeaderTypography = styled(Typography)(() => ({
```
> frontend/src/pages/editTeamPage/EditTeamPage.style.ts


> frontend/src/pages/createTeamPage/CreateTeamPage.style.ts

одинаковые стили, можно сделать отдельный компонент, в который передается props title

```
const Header: FC = ({title}) => ( // или children, например 
  <CustomHeaderTypography>{title}</CustomHeaderTypography>
);
```

Yauhenia-Zhdanovich commented on frontend/src/pages/statisticsPage/StatisticsPage.tsx:
```diff
@@ -1,31 +1,41 @@
 const StatisticsPage: FC = () => {
+  const { pathname } = useLocation();
+
+  const setActiveTabFromPathname = useCallback(() => {
+    return pathname !== RoutePaths.statistics_players
+      ? StatisticsPageEnum.TEAMS
+      : StatisticsPageEnum.PLAYERS;
+  }, [pathname]);
+
```
```suggestion
  const setActiveTabFromPathname = useCallback(() => {
    return pathname === RoutePaths.statistics_players
      ? StatisticsPageEnum.PLAYERS
      : StatisticsPageEnum.TEAMS;
  }, [pathname]);

```

Yauhenia-Zhdanovich commented on frontend/src/shared/config/routeConfig.tsx:
```diff
@@ -8,12 +15,24 @@ export type AppRoutesProps = RouteProps & {
+  [AppRoutes.STATISTICS_TEAMS]: "/statistics/teams",
+  [AppRoutes.STATISTICS_PLAYERS]: "/statistics/players",
   [AppRoutes.FANTASY_TEAMS]: "/fantasy_team",
+  [AppRoutes.CREATE_TEAM]: "/fantasy_team/create_team",
```
обычно в `url` используется `kebab-case`

Yauhenia-Zhdanovich commented on frontend/src/widgets/Navbar/Navbar.tsx:
```diff
@@ -1,54 +1,102 @@
+  isAuth: boolean;
+};
+
+export const Navbar: FC<NavbarProps> = memo(({ isAuth }) => {
+  const { pathname } = useLocation();
+
+  const setActiveTabFromPathname = useCallback(() => {
+    return !navbarStatisticsActiveTabRoutes.includes(pathname)
+      ? NavbarTabsEnum.TEAM_PAGE
+      : NavbarTabsEnum.STATISTICS_PAGE;
+  }, [pathname]);
+
```
```suggestion

const getCurrentTab = (pathname: string) => navbarStatisticsActiveTabRoutes.includes(pathname)
    ? NavbarTabsEnum.STATISTICS_PAGE
    : NavbarTabsEnum.TEAM_PAGE;

export const Navbar: FC<NavbarProps> = memo(({ isAuth }) => {
  const { pathname } = useLocation();
  const [activeTab, setActiveTab] = useState<NavbarTabsEnum>(getCurrentTab(pathname));

  useEffect(() => {
    setActiveTab(getCurrentTab(pathname));
  }, [pathname])

  const handleTabChange = (_: unknown, newValue: NavbarTabsEnum) => setActiveTab(newValue);


```

useCallback тут лишний

Yauhenia-Zhdanovich commented on frontend/src/widgets/Navbar/Navbar.tsx:
```diff
@@ -1,54 +1,102 @@
+        <CustomLink to={RoutePaths.statistics_teams}>
+          <Typography component={"h1"}>Fantasy league</Typography>
+        </CustomLink>
+        {!navbarNoTabsRoutes.includes(pathname) &&
```
```suggestion
       const isSignupOrLogin = navbarNoTabsRoutes.includes(pathname)
       ....
        {!isSignupOrLogin &&
```

DaniilShuliyatev committed:
corrections based on PR review
fixed the not_found router, added the component displayed on this page
fixed the display of empty and /statistics routes by redirecting to /statistics/teams
added redirectable routes in AppRouter component
added CustomHeaderTypography ui component
changed Navbar tabsVisibility logic from finding unmapped paths to finding mapped paths
